src/azure-cli-core/azure/cli/core/tests/test_util.py

```diff
@@ -198,6 +198,140 @@ def test_can_launch_browser(self, webbrowser_get_mock, get_platform_mock):
             result = can_launch_browser()
             self.assertFalse(result)
 
+    def test_configured_default_setter(self):
+        config = mock.MagicMock()
+        config.use_local_config = None
+        with ConfiguredDefaultSetter(config, True):
+            self.assertEqual(config.use_local_config, True)
+        self.assertIsNone(config.use_local_config)
+
+        config.use_local_config = True
+        with ConfiguredDefaultSetter(config, False):
+            self.assertEqual(config.use_local_config, False)
+        self.assertTrue(config.use_local_config)
+
+    @mock.patch('azure.cli.core.__version__', '7.8.9')
+    def test_get_az_user_agent(self):
+        actual = get_az_user_agent()
+        self.assertEqual(actual, 'AZURECLI/7.8.9')
```

Comment:
https://github.com/Azure/azure-cli/pull/12717 can use this test method to verify whether the install is added to UA correctly.